api/api_sources/sources/server/modules/request.access/requestAccess.routes.ts

```diff
@@ -119,8 +119,15 @@ class RequestAccessRouteController extends RouteController {
             }
             return [200, data];
         } else {
+            const allPendingRequests: RequestAccess[] = await this.dataController.all({
+                status: 0,
+                requester_user_id: req.user,
```

Comment:
This portion is proper or we need to create the query filter like this one [requester_user_id: req.user.user_id]. I am just curious.

Reply:
Good catch, Pushan. I just noticed that it is being weird for some reason. I'll do some testing locally and make changes

Reply:
This is super weird actually. In some cases, If the query doesn't match, it returns the first record instead of `undefined`. 
**works when:**
- requester: req.user

**won't wont:**
- requester: req.user.user_id
- requester_user_id: req.user
- requester_user_id: req.user.user_id

Reply:
Yes, I guessed so, because, TypeORM expect an object in its query builder.

Reply:
requeter.user_id: req.user.user_id